src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -81,41 +81,54 @@ public static Task CopyToEndAsync(this IPipeReader input, Stream stream)
         {
             try
             {
-                await input.CopyToAsync(stream, bufferSize, cancellationToken);
+                // TODO: Use bufferSize argument
+                while (!cancellationToken.IsCancellationRequested)
```

Comment:
Registering to the token can be done in another PR, but it still shouldn't be possible to exit this method without Completing the IPipeReader. The easiest solution for now is to remove the CancellationToken overload until it's properly supported.

As I mentioned in my comment in the "CopyFromPipeNoEnd()" test, it doesn't make sense to have separate CopyToAsync and CopyToEndAsync methods in this direction since you always finish reading the IPipeReader in either method (unless Stream.WriteAsync() throws I guess). Might as well call IPipeReader .Complete().

Even when Stream.WriteAsync() throws, a lot of the time the producer will no longer be producing data making it unable to observe the exception as is the case in the "CopyFromPipeToEndObserveStreamThrow" test.